src/nl/hannahsten/texifyidea/settings/conventions/TexifyConventionsConfigurable.kt

```diff
@@ -0,0 +1,216 @@
+package nl.hannahsten.texifyidea.settings.conventions
+
+import com.intellij.openapi.options.Configurable
+import com.intellij.openapi.options.SearchableConfigurable
+import com.intellij.openapi.project.Project
+import com.intellij.ui.JBIntSpinner
+import com.intellij.ui.layout.panel
+import com.intellij.util.ui.CollectionItemEditor
+import com.intellij.util.ui.table.TableModelEditor
+import nl.hannahsten.texifyidea.TexifyIcons
+import nl.hannahsten.texifyidea.settings.TexifyConventionsSchemesPanel
+import java.awt.BorderLayout
+import java.awt.Component
+import javax.swing.JComponent
+import javax.swing.JPanel
+import javax.swing.JTable
+import javax.swing.table.DefaultTableCellRenderer
+import javax.swing.table.TableCellRenderer
+
+/**
+ * Configurable for configuring Texify convention settings.
+ *
+ * The goal of this configurable is to make certain parts of the conventions currently hardcoded in the *Magic classes
+ * configurable.
+ *
+ * The Configurable can manage settings for both, project and IDE scope. To manage the settings for the project scope,
+ * the Configurable is registered as a project configurable. Changes made in the UI are transferred to a
+ * [TexifyConventionsSettings] object.
+ */
+class TexifyConventionsConfigurable(project: Project) : SearchableConfigurable, Configurable.VariableProjectAppLevel {
+
+    private val settingsManager: TexifyConventionsSettingsManager =
+        TexifyConventionsSettingsManager.getInstance(project)
+
+    /**
+     * A settings instance to record active changes. This instance is copied to the actual instance whenever settings
+     * should be applied.
+     */
+    private val unsavedSettings: TexifyConventionsSettings = TexifyConventionsSettings()
+
+    private lateinit var schemesPanel: TexifyConventionsSchemesPanel
+    private lateinit var mainPanel: JPanel
+    private lateinit var maxSectionSize: JBIntSpinner
+    private lateinit var labelConventionsTable: TableModelEditor<LabelConvention>
+
+    private fun createConventionsTable(): TableModelEditor<LabelConvention> {
```

Comment:
The table UI still has the plus/minus/copy buttons, but they aren't functional so can be removed?
It would of course be better if they were functional, but for me that's a nice to have and not necessary right now.

Reply:
I couldn't find a way to remove it and also not the copy button. The copy button in particular is added in `createComponent`, but I can't simply override it because to create the component you need the `toolbarDecorator` which is a private member of the `TableModelEditor` unfortunately.

As for editing in general: yes, that would be the non plus ultra, but I outlined in my first comment why that would be a major effort. To add new conventions we would need to know how the added commands/environments work syntactically (e.g. whether the label is a parameter or a command). Since we don't have a way yet to figure it out automatically, the only alternative would be for the user the enter it, which bears the risk that the entered information might be incorrect, leading to incorrect modifications.

Reply:
I removed the buttons, please check if I didn't break anything else

Reply:
> I removed the buttons, please check if I didn't break anything else

Thanks, looks good to me.